### DISCUSSION TITLE
the further details about how to finish the MIMIC-III Waveform Database Matched Subset

### DISCUSSION THREAD

**Discussion by 9rockchen** [MIT-LCP/mimic-code#1383] [MIMIC-III] [answered]
After looking through your paper about MIMIC-III Waveform Database Matched Subset carefully, I have learnt that this database is an outstanding project. I'm very very interested in how to match the Waveform Database with the MIMIC-III Clinical Database. But I couldn't understand it very well. So I wondering if you could provide me with some further information about automated matching.

**Comment by briangow** [comment, marked as answer]
@9rockchen , exactly what you are having difficulty with isn't clear, so please clarify as necessary. Having said that, you can match between the Waveform Database Matched Subset and the Clinical Database on `subject_id` and the date and time information available in both the Waveform and Clinical databases. 

These sections in the [Waveform project description](https://physionet.org/content/mimic3wdb-matched/1.0/) provide relevant details:

- The name of each matched waveform record is of the form matched/pXX/pXXNNNN/pXXNNNN-YYYY-MM-DD-hh-mm, where XXNNNN is the matching MIMIC-III Clinical Database Subject_ID, and YYYY, MM, DD, hh, and mm are the surrogate year, month (01-12), and day (01-31), and the real hour (00-23) and minute (00-59), derived from the starting date and time of day of the record.
- The master waveform header file for the first stay ([p044083-2112-05-04-19-50.hea](https://physionet.org/content/mimic3wdb-matched/1.0/p04/p044083/p044083-2112-05-04-19-50.hea)) indicates that the record is 20342033 sample intervals (about 45 hours) in length, and begins at 19:50 on May 4, 2112. This date, as with all dates in MIMIC-III, has been anonymized by shifting it by a random number of days into the future.
- The [ICUSTAYS](https://physionet.org/content/mimiciii-demo/1.4/ICUSTAYS.csv) table shows us that he was admitted once to the SICU and twice to the CCU: The first of these admissions corresponds to the waveform record above, as indicated by the date (2112-05-04). Note that the starting and ending date and time of the waveform record will not always match the precise admission or discharge time.

The [WFDB tools ](https://wfdb.io/)make it easy to work with the waveforms. You can download waveforms to temporary memory. You can also extract the date information from the header. For example to get the date from the header for the file in the example you can do this in Python:

```
import wfdb

hd = wfdb.rdheader('p044083-2112-05-04-19-50', pn_dir='mimic3wdb-matched/p04/p044083/')
date = hd.base_date
```
